avionics/common/include/states/ascent_to_apogee.h

```diff
@@ -55,20 +56,35 @@ class AscentToApogee : public IState {
     void onEntry() override {
         apogee_checks = 0;
         mach_checks = 0;
+        initial_time = 0;
         // Awkward to pass in first value - 0-initialize works with checks.
         last_alt = 0;
     }
 
+    StateId mach_timeout() {
+        if (mach_checks == MACH_LOCK_CHECKS_) {
+            initial_time = Hal::millis();
+            return mach_lock_id_;
+        }
+        else if (Hal::millis() - initial_time > MACH_LOCK_TIME_) {
+            mach_checks = 0;
+            return StateId::ASCENT_TO_APOGEE;
+        }
+        return mach_lock_id_;
+    }
+
   private:
     const uint8_t APOGEE_CHECKS_;
     const uint8_t MACH_LOCK_CHECKS_;
     const float MACH_LOCK_VELOCITY_;
+    const uint32_t MACH_LOCK_TIME_;
     const StateId post_apogee_id_;
     const StateId mach_lock_id_;
     Ignitor &drogue_ignitor_;
 
     uint8_t apogee_checks{0};
     uint8_t mach_checks{0};
+    uint32_t initial_time{0};
     float last_alt{0};
```

Comment:
Probably my fault for not going through and using a consistent naming standard earlier, but private member variables should be indicated with a underscore suffix `initial_time_`; if you could change the other private member variables here while you're at it that'd be appreciated